### PR TITLE
Adding Jenkins ENV to Integration Test Job

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
Adding support for Jenkins AWS Role and Session to ENV for Integration Test Job
  
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gaiksaya — APPROVED** [review]
nit: We should be specific with the naming. `AWS_ROLE_ARN` is very generic. 
Maybe something like `AWS_TEST_ROLE_ARN`. I think we are good for now since this jenkinsfile is under integtest but for future

**Review by owaiskazi19 — DISMISSED** [review]

File: bundle-workflow/jenkins_workflow/integtest/Jenkinsfile [hunk 1]
```diff
@@ -1,5 +1,9 @@
 pipeline {
     agent none
+    environment {
+        AWS_ROLE_ARN = "arn:aws:iam::${AWS_ACCOUNT_PUBLIC}:role/opensearch-test"
+        AWS_ROLE_SESSION_NAME = "jenkins-session"
```
Comment (owaiskazi19) [inline bundle-workflow/jenkins_workflow/integtest/Jenkinsfile:5]: Nit: Can we name it as `jenkins-integration-test-session`

**Comment by saratvemulapalli** [comment]
> nit: We should be specific with the naming. `AWS_ROLE_ARN` is very generic.
> Maybe something like `AWS_TEST_ROLE_ARN`. I think we are good for now since this jenkinsfile is under integtest but for future

Yeah the downstream dependency needed it that way.

**Comment by gaiksaya** [comment]
> > nit: We should be specific with the naming. `AWS_ROLE_ARN` is very generic.
> > Maybe something like `AWS_TEST_ROLE_ARN`. I think we are good for now since this jenkinsfile is under integtest but for future
> 
> Yeah the downstream dependency needed it that way.
> Ref: https://github.com/opensearch-project/opensearch-build/tree/main/bundle-workflow#integration-tests

Yeah talking about the overall codebase here :)

**Review by gaiksaya — APPROVED** [review]

**Review by owaiskazi19 — APPROVED** [review]

**Comment by setiah** [comment]
> > > nit: We should be specific with the naming. `AWS_ROLE_ARN` is very generic.
> > > Maybe something like `AWS_TEST_ROLE_ARN`. I think we are good for now since this jenkinsfile is under integtest but for future
> > 
> > 
> > Yeah the downstream dependency needed it that way.
> > Ref: https://github.com/opensearch-project/opensearch-build/tree/main/bundle-workflow#integration-tests
> 
> Yeah talking about the overall codebase here :)

Yeah, there has been some back n forth on the naming - https://github.com/opensearch-project/opensearch-build/pull/367#discussion_r700644640. 
I think we should rename this to TEST_WORKFLOW_ROLE_ARN and refactor the S3Bucket to read it always from the caller, which in this case would be test workflow components.

**Comment by gaiksaya** [comment]
> > > > nit: We should be specific with the naming. `AWS_ROLE_ARN` is very generic.
> > > > Maybe something like `AWS_TEST_ROLE_ARN`. I think we are good for now since this jenkinsfile is under integtest but for future
> > > 
> > > 
> > > Yeah the downstream dependency needed it that way.
> > > Ref: https://github.com/opensearch-project/opensearch-build/tree/main/bundle-workflow#integration-tests
> > 
> > 
> > Yeah talking about the overall codebase here :)
> 
> Yeah, there has been some back n forth on the naming - [#367 (comment)](https://github.com/opensearch-project/opensearch-build/pull/367#discussion_r700644640).
> I think we should rename this to TEST_WORKFLOW_ROLE_ARN and refactor the S3Bucket to read it always from the caller, which in this case would be test workflow components.

That conversation makes sense. Thanks!

**Pull request merged**